src/pages/index.js

```diff
@@ -67,36 +100,99 @@ function handleImageClick(evt) {
   };
   imageModal.open(imageData);
 }
+function avatarButtonClickHandler() {
+  popupAvatarForm.open();
+}
+
+function avatarFormSubmitHandler(avatarLink) {
+  const avatarUrl = avatarLink.url;
+  return api
+    .patchAvatar(avatarUrl)
+    .then(() => {
+      user.setUserAvatar(avatarUrl);
+    })
+    .catch((err) => {
+      console.error(err);
+    });
+}
+
 function editButtonClickHandler() {
-  const currentUserInfo = userInfo.getUserInfo();
-  profileFormName.value = currentUserInfo.name;
-  profileFormDesc.value = currentUserInfo.description;
+  const currentUser = user.getUserInfo();
+  profileFormName.value = currentUser.name;
+  profileFormDesc.value = currentUser.about;
   profileFormValidator.enableFormButton();
   popupProfileForm.open();
 }
 function profileFormSubmitHandler(inputValues) {
-  userInfo.setUserInfo(inputValues);
+  return api
+    .patchUserInfo(inputValues)
+    .then(user.setUserInfo(inputValues))
+    .catch((err) => {
+      console.error(err);
+    });
 }
+
 function addCardButtonClickHandler() {
   popupAddCardForm.open();
 }
-function handleAddCardFormSubmit(inputValues) {
+function addCardFormSubmitHandler(inputValues) {
   const cardData = {
     name: inputValues.title,
     link: inputValues.url,
   };
-  section.addItem(createCard(cardData));
-  popupAddCardForm.close();
-  addCardFormValidator.disableFormButton();
-  addCardForm.reset();
+
+  return api
+    .postNewCard(cardData)
+    .then((newCard) => {
+      section.addItem(createCard(newCard));
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      popupAddCardForm.close();
+      addCardFormValidator.disableFormButton();
+      addCardForm.reset();
+    });
+}
+function handleDeleteButtonClick(cardId) {
+  currentCardId.id = cardId;
+  popupDelete.open();
+}
+
+function handleDeleteSubmit() {
+  return api
+    .deleteCard(currentCardId.id)
+    .then(() => {
+      let cardToRemove = document.getElementById(currentCardId.id);
+      cardToRemove.remove();
+      cardToRemove = null;
+    })
+    .catch((err) => {
+      console.error(err);
```

Comment:
Good job that you `catch` possible errors at the end of server requests. You can simplify it with `.catch(console.error);` because `console.error` is meant to log errors

Reply:
I'm leaving this part as is because the only way I figured out to stop the form from closing on error was to return the promise in the catch, as shown in the new pull request I'm submitting. If this is improper syntax then I can correct it.
![Screenshot 2024-07-04 154238](https://github.com/Bree-Mass/se_project_aroundtheus/assets/166764507/f2aa22f3-7b54-48e3-a7a9-659eeaab5e8f)